### PR TITLE
[9.0][IMP] purchase_request_to_rfq: cancelating procurements in hands of purchase_request

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
The aim of this PR is to modify the cancellation of purchase orders (POs) to not propagate the cancel signal to related procurements and the chain of moves if the origin of a PO line is a purchase request. 

This scenario only applies if you have also `purchase_request_procurement`installed. To complete the cancellation as it is done by default the user has to also cancel the purchase request with the fixes proposed in https://github.com/OCA/purchase-workflow/pull/398

### CODE REVIEW

**Review by AaronHForgeFlow — COMMENTED** [review]
Tested on runbot. Works smoothly :cool: 
Would you add some tests?

**Comment by LoisRForgeFlow** [comment]
@aheficent Yes, it is in my to do list :sweat_smile:

**Review by JordiBForgeFlow — APPROVED** [review]

**Comment by JordiBForgeFlow** [comment]
@gva-acsone @benwillig Do you agree? Once merged we plan to inmediately forward the proposed changes to 10.0

**Comment by lreficent** [comment]
Also related to issue https://github.com/OCA/purchase-workflow/pull/419

**Comment by LoisRForgeFlow** [comment]
Also related to issue https://github.com/OCA/purchase-workflow/issues/396

**Comment by LoisRForgeFlow** [comment]
To Do: 
* [ ] : Add more tests.
* [x] : Bump version.
* [x] : Update README.

**Comment by LoisRForgeFlow** [comment]
Test are working in my local, don't know why they are failling in this built :confused:

File: purchase_request_to_rfq/models/purchase_order.py
Comment (benwillig) [inline purchase_request_to_rfq/models/purchase_order.py]: We loop twice to test the same condition. Maybe we can use following?

```python
remaining_po = self - po_from_request
```

File: purchase_request_to_rfq/models/purchase_order.py
Comment (benwillig) [inline purchase_request_to_rfq/models/purchase_order.py]: I think that this line should be under
 ```python 
if not self.env.context.get('cancel_procurement'):
```

File: purchase_request_to_rfq/models/purchase_order.py
Comment (benwillig) [inline purchase_request_to_rfq/models/purchase_order.py:138]: You loop 3 times on ```procurements```. I don't know if the performance loss is significative. What do you think @sbidoul?

File: purchase_request_to_rfq/models/purchase_order.py
Comment (LoisRForgeFlow) [reply purchase_request_to_rfq/models/purchase_order.py]: that's a good one, I'll add!

File: purchase_request_to_rfq/models/purchase_order.py
Comment (LoisRForgeFlow) [reply purchase_request_to_rfq/models/purchase_order.py]: sure, you're right!

**Comment by MiquelRForgeFlow** [comment]
The tests are failing...

**Comment by LoisRForgeFlow** [comment]
@mreficent I know they are, but at the moment I had to keep this aside. As soon as I have some spare time I will come back to this. If you can't wait, feel free to update the tests yourself, it would be appreciated. :smiley:

**Comment by lreficent** [comment]
This is very weird, I was trying to fix the tests but they were actually running OK in my local. Once pushed, indeed, travis fails because of tests. Looking into it in a [non-practical way](https://github.com/OCA/purchase-workflow/pull/419/commits/25ea8644339fa23c626405749a30bd6d4f3a6231) I found that a procurement is not creating any PO line, which should (totally standard) and in fact occurs even if I do manually through OCA runbot... I really have no idea how to get to the issue and adding commits and waiting for CI is definitely not efficient way to solve it... :sweat_smile:

**Comment by LoisRForgeFlow** [comment]
@aheficent @benwillig travis is now green, could you update your reviews?

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Thank you!

**Pull request merged**
